[ADD][8.0] hr_expense_move_date

andhit-r opened the pull request:
Add move date to expense.
Allows user to force a specific move date for accounting entry

![expense_move_date](https://cloud.githubusercontent.com/assets/1947376/14766800/e6c98998-0a40-11e6-8c55-ea91815711ee.png)

andhit-r committed:
[ADD][8.0] hr_expense_move_date

feketemihai commented:
:+1:

ghost commented:
@andhit-r I don't know how accounting is done in your country, but in Canada, if you have an expense, it must be accounted at the date of the expense. What is the use case behind this module ? When exactly is it permitted by the law to force the account in another period ?

andhit-r commented:
@dufresnedavid same with you, the day expense occured. Things is odoo does not use expense date when create accounting entry. Odoo use date when user click Create Accounting Entry button. Thats why i create this module. So when a back date expense create, user does not have to adjust move date manually.

cyrilgdn commented:
:+1: Tested

andhit-r commented:
@tarzan0820 would you kindly help review this PR?

saltonmassally commented on hr_expense_move_date/README.rst:
```diff
@@ -0,0 +1,83 @@
+2.  Add the path to this repository in your configuration (addons-path)
+3.  Update the module list
+4.  Go to menu *Setting -> Modules -> Local Modules*
+5.  Search For *Employee Expense Move Date*
```
should be "Search For _Expense Move Date_"

saltonmassally commented on hr_expense_move_date/__openerp__.py:
```diff
@@ -0,0 +1,20 @@
+    "website": "https://opensynergy-indonesia.com/",
+    "author": "OpenSynergy Indonesia, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "application": False,
```
application=False is already the default

saltonmassally commented on hr_expense_move_date/models/hr_expense.py:
```diff
@@ -0,0 +1,29 @@
+    _inherit = 'hr.expense.expense'
+
+    move_date = fields.Date(
+        string='Move Date',
```
no need for string="Move Date" here as it can ban directly deduced by odoo here

saltonmassally commented on hr_expense_move_date/models/hr_expense.py:
```diff
@@ -0,0 +1,29 @@
+    @api.model
+    def account_move_get(self, expense_id):
+        res = super(HrExpenseExpense, self).account_move_get(expense_id)
+        expense = self.browse([expense_id])[0]
```
if changed `expense = self.browse(expense_id)` gives you a record so no need to index into a recordset

andhit-r committed:
Modify according @tarzan0820

andhit-r commented:
@tarzan0820 i've push a commit that addressed your reviews except for string parameter on field definition. I understand that Odoo ORM will automatically assign a capitalized string based according field name. But If it's not causing any problem, i would prefer to keep it. Stating it explicitly help readability and easier to explain to other fellow contributor.

ghost commented:
@andhit-r I just tested hr_expense in a vanilla Odoo. As you described, instead of setting the account move date to the date of the expense, Odoo takes the current date when the expense is confirmed.

:+1:

andhit-r commented:
Thank you all reviewers
